.github/workflows/doc_test.yml

```diff
@@ -0,0 +1,35 @@
+name: docs build test
+on:
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout git
+        uses: actions/checkout@v3
+
+      - name: setup python
+        uses: actions/setup-python@v3
+        with:
+          python-version: 3.9
+
+      - name: install gustaf
+        run:  python3 setup.py develop
+
+      - name: Install dependencies
+        run:  pip install -r ./docs/requirements.txt
+
+      - name: install splinepy
```

Comment:
Since I added mock importing of packages, this is not needed anymore. But it also does not hurt.

Reply:
good point

Reply:
um would it still build spline related docs?

Reply:
Yes, it thinks it can import `splinepy`, and therefore the spline sub package is fully loaded. The help will even show type annotations if they are defined in `splinepy`. Since all imported objects are mocked (similar to what I do, the object is loaded but has no functionality)

Reply:
ok, I'll rm

Reply:
that is not what is holding up the review I tried to understand what happens in the vertices changes. This was just an fyi